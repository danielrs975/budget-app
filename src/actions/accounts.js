/**
 * This script contain all the actions about the
 * accounts information
 * @author Daniel Rodriguez
 */
import uuid from 'uuid';

/**
 * This action add an account to the database
 * @param {*} account all the info about the account
 */
export const addAccount = (account) => ({
	type    : 'ADD_ACCOUNT',
	account
});

export const startAddAccount = (account) => {
	return (dispatch, getState) => {
		dispatch(addAccount({ id: uuid(), ...account }));
		const accounts = getState().accounts;
		localStorage.setItem('accounts', JSON.stringify(accounts));
	};
};

/**
 * This action is set in the state the accounts that are saved
 * in the db
 * @param {*} accounts The accounts to set in the state
 */
export const setAccount = (accounts) => ({
	type     : 'SET_ACCOUNTS',
	accounts
});

export const startSetAccount = () => {
	return (dispatch) => {
		const accounts = JSON.parse(localStorage.getItem('accounts'));
		dispatch(setAccount(accounts || []));
	};
};
