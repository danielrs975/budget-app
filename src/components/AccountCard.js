/**
 * This component contain a short information about an account
 * @author Daniel Rodriguez
 */
import React from 'react';

const AccountCard = ({ name, amount }) => (
	<div className="card-layout">
		<h3>{name}</h3>
		<p>{amount / 100}</p>
	</div>
);

export default AccountCard;
