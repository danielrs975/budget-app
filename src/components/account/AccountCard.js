/**
 * This component contain a short information about an account
 * @author Daniel Rodriguez
 */
import React from 'react';
import { Link } from 'react-router-dom';

const AccountCard = ({ name, amount, id }) => (
	<div className="card-layout">
		<Link to={`account/${id}`}><h3>{name}</h3></Link>
		<p>{amount / 100}</p>
	</div>
);

export default AccountCard;
