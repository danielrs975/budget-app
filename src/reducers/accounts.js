/**
 * This is the reducer for the accounts
 * information
 * @author Daniel Rodriguez
 */

export default (state = [], action) => {
	switch (action.type) {
		case 'ADD_ACCOUNT':
			return [ ...state, action.account ];
		default:
			return state;
	}
};
