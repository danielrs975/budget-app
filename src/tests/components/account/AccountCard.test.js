import React from 'react';
import { shallow } from 'enzyme';
import accounts from '../../fixtures/accounts';
import AccountCard from '../../../components/account/AccountCard';

test('should render AccountCard correctly', () => {
	const wrapper = shallow(<AccountCard account={accounts[0]} />);
	expect(wrapper).toMatchSnapshot();
});
