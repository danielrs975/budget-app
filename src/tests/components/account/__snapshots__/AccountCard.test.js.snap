// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AccountCard correctly 1`] = `
<div
  className="card-layout"
>
  <[object Object]
    to="account/undefined"
  >
    <h3 />
  </[object Object]>
  <p>
    NaN
  </p>
</div>
`;
